backend/editor/api.py

```diff
@@ -369,6 +380,34 @@ async def import_from_github(request: Request, branch: str, taxonomy_name: str):
     return result
 
 
+@app.post("/{branch}/upload")
+async def upload_taxonomy(
+    request: Request,
+    branch: str,
+    file: UploadFile,
+    description: str = Form(...)
+):
+    """
+    Upload taxonomy file to be parsed
+    """
+    # use the file name as the taxonomy name
+    taxonomy = TaxonomyGraph(branch, file.filename.split(".")[0])
```

Comment:
I think it's best if we disregard the filename, and allow user to select a taxonomy through the UI, because we depend on the user to name the file correctly.